### PR TITLE
fix: Apply isReactionEnabled from SendbirdProvider to Channel

### PR DESCRIPTION
**Pull request by HoonBaek**
## Description Of Changes

Apply the `isReactionEnabled` props from the SendbirdProvider to Channel component
* Add `isReactionEnabled` props into the SendBirdStateConfig (internal interface)
* Keep sync the properties of the interface ChannelContextProps

## Types Of Changes

What types of changes does your code introduce to this project?
Put an `x` in the boxes that apply_

- [x] Bugfix
- [ ] New feature
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance (ex) Prettier)
- [ ] Build configuration
- [ ] Improvement (refactor code)

### CODE REVIEW

File: scripts/index_d_ts [hunk 1]
```diff
@@ -466,9 +467,8 @@ declare module "SendbirdUIKitGlobal" {
     onSearchClick?(): void;
     replyType?: ReplyType;
     queries?: ChannelQueries;
-    renderUserProfile?: (props: RenderUserProfileProps) => React.ReactNode | React.ReactElement;
+    renderUserProfile?: (props: RenderUserProfileProps) => React.ReactElement;
```
Comment (sravan-s) [inline scripts/index_d_ts:470]: This change necessary?

File: scripts/index_d_ts [hunk 2]
```diff
@@ -466,9 +467,8 @@ declare module "SendbirdUIKitGlobal" {
...
     disableUserProfile?: boolean;
-    renderUserMentionItem?: (props: { user: User }) => JSX.Element;
```
Comment (sravan-s) [inline scripts/index_d_ts:471]: Why remove this?

**Review by sravan-s — APPROVED** [review]

File: scripts/index_d_ts (hunk 2 above)
Comment (HoonBaek) [reply scripts/index_d_ts:471]: This is our legacy, we don't get this input on the Channel component

File: scripts/index_d_ts (hunk 1 above)
Comment (HoonBaek) [reply scripts/index_d_ts:470]: ah, seems this is overlapped while I copy the interface from the Channel file, let me make it back

**Pull request merged**